gemsfx/src/main/java/com/dlsc/gemsfx/CalendarPicker.java

```diff
@@ -179,4 +177,31 @@ public final ObjectProperty<StringConverter<LocalDate>> converterProperty() {
     public final void setConverter(StringConverter<LocalDate> converter) {
         this.converter.set(converter);
     }
+
+    private final ObjectProperty<Callback<LocalDate,Boolean>> dateFilter = new SimpleObjectProperty<>(this, "dateFilter");
+
+    public Callback<LocalDate, Boolean> getDateFilter() {
+        return dateFilter.get();
+    }
+
+    /**
+     * A property to define a filter for determining which dates in the calendar can be selected.
+     * This filter is applied to each date displayed in the calendar. If the filter returns true for
+     * a given date, that date will be selectable (i.e., it passes the filter). If the filter returns
+     * false, the date will be disabled and cannot be selected. This property is particularly useful
+     * for scenarios where only specific dates should be available for selection based on custom
+     * logic, such as business rules, holidays, or availability.
+     * <p>
+     * When SelectionMode is DATE_RANGE, disabled dates can be included within the selected range.
```

Comment:
use {@link Mode#DATE_RANGE} to create a link to the enum value. Same for the selection mode. Use {@link #selectionModeProperty}.